docs/sphinxext/apigen.py

```diff
@@ -175,34 +176,34 @@ def _uri2path(self, uri):
         >>> import sphinx
         >>> modpath = sphinx.__path__[0]
         >>> res = docwriter._uri2path('sphinx.builder')
-        >>> res == os.path.join(modpath, 'builder.py')
+        >>> res == pathlib.PurePath(modpath, 'builder.py')
         True
         >>> res = docwriter._uri2path('sphinx')
-        >>> res == os.path.join(modpath, '__init__.py')
+        >>> res == pathlib.PurePath(modpath, '__init__.py')
         True
         >>> docwriter._uri2path('sphinx.does_not_exist')
 
         '''
         if uri == self.package_name:
-            return os.path.join(self.root_path, '__init__.py')
-        path = uri.replace('.', os.path.sep)
-        path = path.replace(self.package_name + os.path.sep, '')
-        path = os.path.join(self.root_path, path)
+            return pathlib.PurePath(self.root_path, '__init__.py')
+        path = uri.replace('.', pathlib.os.sep)
+        path = path.replace(self.package_name + pathlib.os.sep, '')
+        path = pathlib.PurePath(self.root_path, path)
         # XXX maybe check for extensions as well?
-        if os.path.exists(path + '.py'): # file
+        if pathlib.Path(path + '.py').exists(): # file
             path += '.py'
-        elif os.path.exists(os.path.join(path, '__init__.py')):
-            path = os.path.join(path, '__init__.py')
+        elif pathlib.Path(pathlib.PurePath(path, '__init__.py')).exists():
+            path = pathlib.PurePath(path, '__init__.py')
         else:
             return None
         return path
 
     def _path2uri(self, dirpath):
         ''' Convert directory path to uri '''
         relpath = dirpath.replace(self.root_path, self.package_name)
```

Comment:
The build fails here because you pass a PurePath object to this function, but PurePaths don't have I/O operations (replace() is one of those). You want to use Path instead when accessing real files.